test/spec-directory/cleanup.test.ts

```diff
@@ -0,0 +1,30 @@
+import { withAsyncCleanup } from "../../lib-js/spec-directory/cleanup"
+
+describe("withAsyncCleanup", () => {
+  it("does the cleanup function on normal exit", async () => {
+    const mock = jest.fn()
+    await withAsyncCleanup(mock, async () => {
+      // do nothing
+    })
+    expect(mock).toHaveBeenCalled()
+  })
+
+  it("does the cleanup on an error", async () => {
+    try {
+      const mock = jest.fn()
+      await withAsyncCleanup(mock, async () => {
+        throw new Error("YEET")
+      })
+      expect(mock).toHaveBeenCalled()
+    } catch (e) {}
+  })
+
+  it.skip("does the cleanup on a SIGINT", async () => {
+    // const mockExit = jest.spyOn(process, "exit").mockImplementation()
+    const mock = jest.fn()
+    await withAsyncCleanup(mock, async () => {
+      process.kill(process.pid, "SIGINT")
+    })
+    expect(mock).toHaveBeenCalled()
```

Comment:
How is this not just killing the process?

Reply:
…it does, which is why this test is skipped for now. 😓

Probably the best way to test this would be to call the `withAsyncCleanup` function from a script, then exec that script from the test and check that cleanup occurs. But that makes the tests run a lot slower (because of the overhead of ts-node compilation).